File: AppTemplate/.plugins/android/com/pkg/logger/LoggerPlugin.java
package com.juicepackage.logger;

import android.content.Context;
import android.content.Intent;
import android.util.Log;
import com.juiceplugin.SweetJuicePlugin;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * LoggerPlugin routes logs from Go to Android's Logcat.
 */
public class LoggerPlugin implements SweetJuicePlugin {
    @Override
    public String getDomain() { return "logger"; }

    @Override
    public void onAttach(Context context) {}

    @Override
    public String handleAction(String action, String jsonArgsPayload) {
        if ("log".equals(action)) {
            try {
                JSONObject args = new JSONObject(jsonArgsPayload);
                String tag = args.optString("tag", "SweetJuiceGo");
                String level = args.optString("level", "INFO");
                String message = args.optString("message", "");

                switch (level) {
                    case "DEBUG":
                        Log.d(tag, message);
                        break;
                    case "WARN":
                        Log.w(tag, message);
                        break;
                    case "ERROR":
                        Log.e(tag, message);
                        break;
                    case "INFO":
                    default:
                        Log.i(tag, message);
                        break;
                }
                return "{\"status\":\"ok\"}";
            } catch (JSONException e) {
                return "{\"error\":\"Invalid JSON\"}";
            }
        }
        return "{\"error\":\"Unknown action\"}";
    }

    @Override public void onActivityResult(int req, int res, Intent d) {}
    @Override public void onRequestPermissionsResult(int req, String[] p, int[] res) {}
}

File: AppTemplate/native/ios/Sources/GenericApp/Plugins/Logger/LoggerPlugin.swift
import Foundation
import UIKit

@MainActor
public class LoggerPlugin: SweetJuicePlugin {
    public init() {}

    public func getDomain() -> String {
        return "logger"
    }

    public func onAttach(container: UIViewController) {}

    public func handleAction(action: String, jsonArgs: String) -> String {
        guard let data = jsonArgs.data(using: .utf8),
              let args = try? JSONSerialization.jsonObject(with: data) as? [String: Any] else {
            return "{\"error\":\"Invalid JSON payload\"}"
        }

        let message = args["message"] as? String ?? ""
        let tag = args["tag"] as? String ?? "SweetJuice"

        switch action {
        case "debug":
            print("[\(tag)] DEBUG: \(message)")
        case "info":
            print("[\(tag)] INFO: \(message)")
        case "error":
            print("[\(tag)] ERROR: \(message)")
        default:
            return "{\"error\":\"Unknown action\"}"
        }

        return "{\"status\":\"ok\"}"
    }
}
